osquery/tables/system/darwin/signature.mm

```diff
@@ -105,11 +111,67 @@ void genSignatureForFile(const std::string& path,
         r["signed"] = INTEGER(1);
         r["identifier"] = stringFromCFString(ident);
 
-        // TODO(andrew-d): can get more information from the signature here?
+        // Get CDHash
+        r["cdhash"] = "";
+        CFTypeRef hashInfo = CFDictionaryGetValue(codeInfo, kSecCodeInfoUnique);
+        if (hashInfo) {
+          r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
+          CFDataRef hash = CFDataRef(hashInfo);
+          // Get the SHA-1 bytes
+          std::stringstream ss;
+          auto bytes = CFDataGetBytePtr(hash);
```

Comment:
You can do: `auto bytes = CFDataGetBytePtr((CFDataRef)hashInfo);` and remove the intermediate `hash` variable.